src/modules/0.0.11/aws_iam/index.ts

```diff
@@ -92,8 +118,18 @@ export const AwsIamModule: Module2 = new Module2({
             const b = cloudRecord;
             let update = false;
             let updatedRecord = { ...cloudRecord };
-            if (!Object.is(e.assumeRolePolicyDocument, b.assumeRolePolicyDocument)) {
-              await client.updateRoleAssumePolicy(e.roleName, e.assumeRolePolicyDocument);
+            if (!AwsIamModule.utils.rolePolicyComparison(e.assumeRolePolicyDocument, b.assumeRolePolicyDocument)) {
+              await client.updateRoleAssumePolicy(e.roleName, JSON.stringify(e.assumeRolePolicyDocument));
+              try {
+                const allowEc2Service = AwsIamModule.utils.allowEc2Service(e);
+                if (allowEc2Service) {
+                  await client.createInstanceProfile(e.roleName);
+                  await client.attachRoleToInstanceProfile(e.roleName);
+                }
+              } catch (err: any) {
+                // Do not throw if for any reason this does not work
+                logger.warn(err.message ?? 'Error attaching instance profile on role update');
+              }
```

Comment:
Same here, this seems weird to me?

Reply:
Here because if the role already exists, and we are updating adding something else to the policy, the attachment already exists and would fail on duplication. But, If we are changing the policy role to support now ec2, we want to create the attachment.

Reply:
but now that I think of it it is not being handled the case where it was ec2 and now it is being something else, we need to detach the instance profile. I'll work on that